src/main/java/digital/slovensko/autogram/core/AutogramBatchStartCallback.java

```diff
@@ -0,0 +1,44 @@
+package digital.slovensko.autogram.core;
+
+import java.util.function.Consumer;
+
+import digital.slovensko.autogram.core.errors.AutogramException;
+import digital.slovensko.autogram.ui.gui.GUI;
+import digital.slovensko.autogram.util.Logging;
+
+public class AutogramBatchStartCallback implements Consumer<SigningKey> {
+
+    private final Batch batch;
+    private final BatchResponder responder;
+
+    public AutogramBatchStartCallback(Batch batch, BatchResponder responder) {
+        this.batch = batch;
+        this.responder = responder;
+    }
+
+    public void accept(SigningKey key) {
+        GUI.assertOnWorkThread();
```

Comment:
Tento check je potrebny este?

Reply:
potrebny nie je, ale vdaka GUI.DEBUG si to vieme zapnut a kontrolovat kde sme ocakavali, ze nieco bude na niektorom threade, mozem odstranit ak ti to pride zbytocne - je to na to aby sme vedeli preklikat a povedat kde sa spusta nieco nespravne/neocakavane